types/underscore/index.d.ts

```diff
@@ -154,7 +154,20 @@ declare module _ {
         : TItem
         : T;
 
-    type PairValueOrAny<T> = T extends [EnumerableKey, infer TValue] ? TValue : any;
+    // if T is an inferrable pair, the value type for the pair
+    // if T is a list, assume that it contains pairs of some type, so any
+    // if T isn't a list, there's no way that it can provide pairs, so never
+    type PairValue<T> =
+        T extends [EnumerableKey, infer TValue, ...any[]] ? TValue
+        : T extends List<any> ? any
```

Comment:
For a little added service:

```suggestion
        : T extends List<infer E> ? E
```

Reply:
Seems reasonable, thanks!

Reply:
Hmm, this does make things a little more interesting when a union type is used; let me know what you think of the new results for the "nested lists" case since I think that will potentially cause new errors for folks in some cases.